src/graphics/Window.h

```diff
@@ -0,0 +1,151 @@
+//
+// Created by ajer on 11.02.19.
+//
+
+#pragma once
+
+class GLFWwindow;
+
+namespace Bald::Graphics {
+
+    /**
+     * @class Window
+     * @brief Implementation of simple Window
+     */
+    class Window {
+    public:
+        /** ========== METHODS ========== */
+
+        /**
+         * @fn                      Window
+         * @param [std::string]     title
+         * @param [int]             width
+         * @param [int]             height
+         * @param [bool             closed - is window closed
+         * @param [bool]            vsync
+         * @brief                   constructor for the window
+         */
+        explicit Window(const char* title = "Window", int width = 800, int height = 600,
+                        bool closed = true, bool VSync = false);
+
+        /**
+         * @fn                      ~Window
+         * @brief                   destructor - terminates the m_Window
+         */
+        ~Window() noexcept { Shutdown(); }
```

Comment:
I think it should be implemented in a .cpp file

Reply:
moved